velox/functions/lib/Re2Functions.cpp

```diff
@@ -378,7 +379,28 @@ class LikeConstantPattern final : public VectorFunction {
  public:
   LikeConstantPattern(StringView pattern, std::optional<char> escapeChar)
       : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
-            RE2::Quiet) {}
+            RE2::Quiet) {
+    pattern_ = pattern;
+    fixedPattern_ = false;
+    prefixPattern_ = false;
+    suffixPattern_ = false;
+
+    if (escapeChar) {
+      return;
+    }
+    if ((fixedPattern_ =
```

Comment:
This code is a bit difficult to read. Maybe it can be simplified as follows:

```
if (!isWildcardInPattern(pattern_)) {
   fixedPattern_ = true;
   return;
}
```